tiledb/sm/query/reader_base.cc

```diff
@@ -1518,12 +1518,15 @@ Status ReaderBase::process_tiles(
   // separately from `copy_names`.
   std::vector<std::string> read_names;
   std::vector<std::string> copy_names;
+  std::vector<std::string> var_size_read_names;
   read_names.reserve(names->size());
   for (const auto& name_pair : *names) {
     const std::string name = name_pair.first;
     const ProcessTileFlags flags = name_pair.second;
     if (flags & ProcessTileFlag::READ) {
       read_names.push_back(name);
+      if (fragment_metadata_.back()->array_schema()->var_size(name))
```

Comment:
Isn't the names here based on the user set buffers? We should use the latest array schema then not the last fragments schema? The latest array schema is what the user can set buffers based on.